gpytorch/lazy/lazy_variable.py

```diff
@@ -1,9 +1,87 @@
 class LazyVariable(object):
     def evaluate(self):
+        """
+        Explicitly evaluates the matrix this LazyVariable represents. This
+        function should return a Variable explicitly wrapping a Tensor storing
+        an exact representation of this LazyVariable.
+        """
         raise NotImplementedError
 
     def add_diag(self, diag):
+        """
+        Adds a diagonal component to this lazy variable in a potentially lazy fashion.
+        That is, if this lazy variable represents some matrix A, add_diag should return
+        a LazyVariable of the same type that represents A+dI.
+
+        Args:
+            - diag (scalar) - diagonal component to add.
+        Returns:
+            - LazyVariable representing this current LazyVariable plus diag*I.
+        """
```

Comment:
If you're just adding docs like this (i.e. not specific to this PR), just commit it directly into master. It's kind of distracting for me to go thorough this stuff in a review, and I end up repeating work like this because I don't know you're doing it.